### PR TITLE
Linked blogs to authors

### LINKED ISSUE #1279: [NG] Link Blog Author to Member Profile
**Desired Feature**
Blog author names should be automatically hyperlinked to their member profile.

---
**Contributor Resources**
The `layer5` repo contains two websites. The current generation and the next-generation of the layer5.io site.

If the `layer5-ng` label is absent on this issue, then this issue pertains to the current generation of the layer5.io website, which uses Jekyll and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- See [`layer5` contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md)

If the `layer5-ng` label is present on this issue, then this issue pertains to the next-generation of the layer5.io website, which uses Gatsby, Strapi, and GitHub Pages. Site content is found under the [`layer5-ng` branch](https://github.com/layer5io/layer5/tree/layer5-ng).
- See [`layer5-ng` contributing instructions](https://github.com/layer5io/layer5/blob/layer5-ng/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by OmkarPh**
Signed-off-by: OmkarPh <omkarphansopkar@gmail.com>

**Description**

This PR fixes #1279

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by OmkarPh** [comment]
How can i trigger deploy with newer commits?

**Comment by Tanuj22** [comment]
> How can i trigger deploy with newer commits?

It's automatic

**Comment by kishore007k** [comment]
Wouldn't it be nice to have the author's name on the `blogs` page to link to their corresponding member profile!
Capture3

**Comment by OmkarPh** [comment]
> > How can i trigger deploy with newer commits?
> 
> It's automatic 

The preview isn't reflecting changes 😕
Also, you can see, I push last commit just 5 to 10 mins before, but preview and bot msg was edited 1hr ago

**Comment by Nikhil-Ladha** [comment]
> Wouldn't it be nice to have the author's name on the `blogs` page to link to their corresponding member profile!
> Capture3

Yes, it would be. I am working on the blogs page. Will add that in my PR.

**Comment by Nikhil-Ladha** [comment]
This PR can ve merged leaving that part and the blogs code untouched.

**Comment by OmkarPh** [comment]
> Wouldn't it be nice to have the author's name on the `blogs` page to link to their corresponding member profile!
> Capture3
> 

I thought so, but whole card links to blog with hover effect, adding another link might be a bit confusing...

**Comment by OmkarPh** [comment]
> > How can i trigger deploy with newer commits?
> 
> It's automatic 

There was some problem from my end 🤦🏻‍♂️
I pushed in another branch

**Comment by leecalcote** [comment]
@OmkarPh consider a more descriptive PR title, if you would. 

I was not expecting to have another front matter variable added to link to the author's profile. The author variable should be enough to create a linkage system around. What do you think?

**Comment by OmkarPh** [comment]
> @OmkarPh consider a more descriptive PR title, if you would.
> 
> I was not expecting to have another front matter variable added to link to the author's profile. The author variable should be enough to create a linkage system around. What do you think?

That was my first bet, but 
https://deploy-preview-1287--layer5ng.netlify.app/blog/announcements/announcing-a-new-line-of-meshmates/ This blog had our meshmates with love, forming a link out of it and checking if that link is correct was another workaround.

https://deploy-preview-1287--layer5ng.netlify.app/blog/istio/service-mesh-istio-patterns-for-multitenancy This blog also has user who's not a part of community members

I should've talked about the approach on slack, pardon me :)

**Comment by Tanuj22** [comment]
@OmkarPh Does using something like this in the author frontmatter work `<Link to `/community/members/dummy`>dummy</Link>`?

**Comment by OmkarPh** [comment]
> @OmkarPh Does using something like this in the author frontmatter work `<Link to = '/community/members/dummy'> dummy </Link>`? 
> 
> Or is there a way to check if a URL exists? 

I saw this now,
Maybe this could've worked too 🤔

If I'd taken this approach what is the best way to convert string to JSX ?

**Comment by OmkarPh** [comment]
By the way, this also adds link to authors in books and wherever Pageheader is used

**Comment by leecalcote** [comment]
@OmkarPh will you add a check to verify the type of collection to which the page belongs? We should be able to apply this link only to pages belonging to the `blog` type collections.

**Comment by OmkarPh** [comment]
I can add, but wouldn't it be good if everything that has a author also has hyperlink?

**Review by kishore007k — DISMISSED** [review]

**Comment by OmkarPh** [comment]
> I can add, but wouldn't it be good if everything that has a author also has hyperlink?

@Lee ^^

File: src/reusecore/PageHeader/index.js [hunk 1]
```diff
@@ -24,14 +45,31 @@ const PageHeader = ({ categories, title, subtitle,  author, thumbnail, path, sub
                         ))}
                         {author && (
                             <>
-                                <h4>By:</h4> <span>{author.name}</span>
+                                <h4>By:</h4> 
+                                { 
+                                    isSlugAvailable ? 
+                                        <Link className="authorLink"
+                                            to={`/community/members/${author.name.toLowerCase().replace(" ","-")}`}>
```
Comment (Tanuj22) [inline src/reusecore/PageHeader/index.js:52]: just use the `slugify` function defined in `utils`

File: src/reusecore/PageHeader/index.js [hunk 2]
```diff
@@ -24,14 +45,31 @@ const PageHeader = ({ categories, title, subtitle,  author, thumbnail, path, sub
...
                             </>
                         )}
                     </div>
                 )}
                 {!categories && author && (
                     <div className="breadcrumbs post">
-                        <h4>By:</h4> <span>{author.name}</span>
+                        <h4>By:</h4> 
+                        { 
+                            isSlugAvailable ?
```
Comment (Tanuj22) [inline src/reusecore/PageHeader/index.js:65]: As this is used in two places make a function out of it and use that

**Comment by leecalcote** [comment]
> > I can add, but wouldn't it be good if everything that has a author also has hyperlink?
> 
> @leecalcote ^^

@OmkarPh great question. `/learn/books` is an example of a set of cards where we do not want the author linked.

**Comment by OmkarPh** [comment]
You mean the books grid or individual book?

File: src/reusecore/PageHeader/index.js (hunk 1 above)
Comment (OmkarPh) [reply src/reusecore/PageHeader/index.js:52]: Oh, thanks

**Comment by leecalcote** [comment]
Books grid, which should be going away and replaced with a carousel.

**Comment by OmkarPh** [comment]
Since we don't provide author object here, query and verification would not run in this case
Screenshot_2020-12-23-11-37-41-71

**Review by Tanuj22 — APPROVED** [review]
LGTM! Thanks @OmkarPh

**Pull request merged**